index.html

```diff
@@ -1830,21 +1901,28 @@ <h2>
               </li>
               <li>
                 <p>
-                  If they do not resolve successfully, then set |notificationResult| to failure.
+                  If they do not resolve successfully, then set |notificationResult| and
+                  |appBadgeResult| to failure and abort these steps.
+                </p>
+              </li>
+              <li>
+                <p>
+                  Set |notificationResult| to |registration|'s [=service worker registration/has
+                  shownotification() been successfully invoked|has `showNotification()` been
+                  successfully invoked=].
                 </p>
               </li>
               <li>
                 <p>
-                  Otherwise, set |notificationResult| to |registration|'s [=service worker
-                  registration/has shownotification() been successfully invoked|has
-                  `showNotification()` been successfully invoked=].
+                  Set |appBadgeResult| to true if {{NavigatorBadge/setAppBadge()}} has been
+                  invoked; otherwise false.
```

Comment:
@marcoscaceres we need a hook for this similar to "has `showNotification()` been successfully invoked". People were not happy with this being defined in a hand-wavy way.